tests/governance.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+import os
+import getpass
+import time
+import logging
+import multiprocessing
+import shutil
+import subprocess
+from random import seed
+import infra.ccf
+import infra.proc
+import infra.jsonrpc
+import infra.notification
+import infra.net
+import e2e_args
+
+from loguru import logger as LOG
+
+
+def wait_for_node_commit_sync(nodes):
+    """
+    Wait for commit level to get in sync on all nodes. This is expected to
+    happen once CFTR has been established, in the absence of new transactions.
+    """
+    for _ in range(3):
+        commits = []
+        for node in nodes:
+            with node.management_client() as c:
+                id = c.request("getCommit", {})
+                commits.append(c.response(id).commit)
+        if [commits[0]] * len(commits) == commits:
+            break
+        time.sleep(1)
+    assert [commits[0]] * len(commits) == commits, "All nodes at the same commit"
+
+
+def run(args):
+    hosts = ["localhost", "localhost"]
+
+    with infra.notification.notification_server(args.notify_server) as notifications:
```

Comment:
I don't think we need this for all end-to-end tests.

Reply:
No, indeed.